app/src/main/java/com/arctouch/codechallenge/mvp/home/MovieRest.java

```diff
@@ -0,0 +1,64 @@
+package com.arctouch.codechallenge.mvp.home;
+
+import com.arctouch.codechallenge.api.TmdbApi;
+import com.arctouch.codechallenge.data.Cache;
+import com.arctouch.codechallenge.model.Genre;
+import com.arctouch.codechallenge.model.Movie;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.schedulers.Schedulers;
+import okhttp3.OkHttpClient;
+import retrofit2.Retrofit;
+import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory;
+import retrofit2.converter.moshi.MoshiConverterFactory;
+
+public class MovieRest {
+    private final HomePresenter mHomePresenter;
+    private List<Movie> listMovie = new ArrayList<>();
+
+    private final TmdbApi api = new Retrofit.Builder()
+            .baseUrl(TmdbApi.URL)
+            .client(new OkHttpClient.Builder().build())
+            .addConverterFactory(MoshiConverterFactory.create())
+            .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
+            .build()
+            .create(TmdbApi.class);
+
+    public MovieRest(HomePresenter aHomePresenter) {
+        mHomePresenter = aHomePresenter;
+    }
+
+    public void callMovies(int currentPage) {
+        api.upcomingMovies(TmdbApi.API_KEY, TmdbApi.DEFAULT_LANGUAGE, (long) currentPage, TmdbApi.DEFAULT_REGION)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(response -> {
```

Comment:
It is a tool that can be used to control the life cycle of an "observable" if the data stream that the "observable" is producing is unlimited. It may remain active forever, it may not be a problem for a server-side application, but it may cause some problems with Android, and the "memory leak" problem is common.